lib/tasks/data/statistics.rake

```diff
@@ -1,402 +1,67 @@
-
-        data
-      end
-
-      bucket_name = 'check-batch-task-statistics'
-      region = 'eu-west-1'
-      begin
-        s3_client = Aws::S3::Client.new(region: region)
-      rescue Aws::Sigv4::Errors::MissingCredentialsError
-        puts "Please provide the AWS credentials."
-        exit 1
-      end
-
-      def cache_team_data(team, header, rows)
-        data = []
-        rows.each do |row|
-          entry = {}
-          header.each_with_index do |column, i|
-            entry[column] = row[i]
-          end
-          data << entry
-        end
-        Rails.cache.write("data:report:#{team.id}", data)
-      end
-
-      def object_uploaded?(s3_client, bucket_name, object_key, file_path)
-        response = s3_client.put_object(
-          bucket: bucket_name,
-          key: object_key,
-          body: File.read(file_path)
-        )
-        if response.etag
-          return true
-        else
-          return false
-        end
-      rescue StandardError => e
-        puts "Error uploading S3 object: #{e.message}"
-        return false
-      end
-
+    task statistics: :environment do |_t|
       old_logger = ActiveRecord::Base.logger
       ActiveRecord::Base.logger = nil
-      args = params.to_a
-      start_year = args[0].to_i
-      start_month = args[1].to_i
-      end_year = args[2].to_i
-      end_year = Time.now.year if end_year == 0
-      end_month = args[3].to_i
-      end_month = Time.now.month if end_month == 0
-      group_by_month = args[4].to_i
-      slugs = args[5].to_s.split('.')
-      if slugs.empty?
-        puts 'Please provide a list of workspace slugs'
-      else
-        outfile = File.open('/tmp/statistics.csv', 'w')
-        header = [
-          'ID',
-          'Org',
-          'Platform',
-          'Language',
-          'Month',
-          'Conversations',
-          'Average number of conversations per day',
-          'Number of messages sent',
-          'Average messages per day',
-          'Unique users',
-          'Returning users',
-          'Searches',
-          'Positive searches',
-          'Negative searches',
-          'Search feedback positive',
-          'Search feedback negative',
-          'Search no feedback',
-          'Valid new requests',
-          'Published native reports',
-          'Published imported reports',
-          'Requests answered with a report',
-          'Reports sent to users',
-          'Unique users who received a report',
-          'Average (median) response time',
-          'Unique newsletters sent',
-          'New newsletter subscriptions',
-          'Newsletter cancellations',
-          'Current subscribers'
-        ]
-        outfile.puts(header.join(','))
 
-        slugs.each do |slug|
-          team = Team.find_by_slug(slug)
-          team_rows = []
+      team_ids = Team.joins(:project_medias).where(project_medias: { user: BotUser.smooch_user }).distinct.pluck(:id)
```

Comment:
@caiosba sanity check (flagging from the Jira comment): do you think this makes sense as a way to detect which workspaces we should be generating statistics for? I took the logic from the previous guard clause on line 50, but inverted it. seems to make sense to me, since I think we want to regenerate historic data.

Reply:
Good catch, yes, that is important because then we can keep generating statistics for workspaces even after they uninstall the tipline bot.